### PR TITLE
Fix attribute error in rms_run.py os.stat result

### PR DESCRIPTION
**Pull request by joakim-hove**
There is a minor syntax error when the rms job is checking the stat output of a target file.

The current rms testing is done with a very basic mock script which does not capture this behaviour at all, it was therefor quite nontrivial to make a proper test for this fix. Since the current code accesses attributes from standard module `os.stat()` incorrectly it should be quite trivial though; and I hope this can be merged.

### CODE REVIEW

**Comment by markusdregi** [comment]
I agree with the code changes. I believe the following test should do the trick:

```python
    def test_run_class_with_existing_target_file(self):
        with TestAreaContext("test_run_existing_target"):
            with open("rms_config.yml", "w") as f:
                f.write("executable:  {}/bin/rms".format(os.getcwd()))

            os.mkdir("run_path")
            os.mkdir("bin")
            os.mkdir("project")
            shutil.copy(os.path.join(self.SOURCE_ROOT, "python/tests/res/fm/rms"), "bin")
            self.monkeypatch.setenv("RMS_SITE_CONFIG", "rms_config.yml")

            target_file = os.path.join(os.getcwd(), "rms_target_file")
            action = {
                "exit_status": 0,
                "target_file": target_file,
            }
            with open("run_path/action.json", "w") as f:
                f.write(json.dumps(action))

            with open(target_file, "w") as f:
                f.write("This is a dummy target file")

            r = RMSRun(0, "project", "workflow", run_path="run_path", target_file=target_file)
            r.run()
```

**Comment by joakim-hove** [comment]
Thank you for the suggested test - I have updated the PR with your suggested test. Would be cool if 7dc50a2 was also cherry picked to the 2.6 release?

**Comment by ertomatic** [comment]
No problem 👍 

We will definitively get it into the 2.6 release. Thanks for the fix :)

**Review by ertomatic — APPROVED** [review]
LGTM.

Please squash your commits before merging.

**Comment by markusdregi** [comment]
No problem 👍

We will definitively get it into the 2.6 release. Thanks for the fix :)

**Comment by joakim-hove** [comment]
> We will definitively get it into the 2.6 release. Thanks for the fix :)

Great - thank you.

> Please squash your commits before merging.

I will happily squash of course - I did keep it in two commits to simplify cherry-picking the actual fix - your call?

**Comment by markusdregi** [comment]
I think I prefer a squash ;)

**Comment by joakim-hove** [comment]
> I think I prefer a squash ;)

Squash it is; observe that one additional change (a `time.sleep(1)` has been introduced in the mock rms binary).

**Comment by markusdregi** [comment]
Observe that it is only the macOS runs that are failing... Maybe that could be a hint?

**Comment by joakim-hove** [comment]
> Observe that it is only the macOS runs that are failing... Maybe that could be a hint?

Ehhh - no not really; I am a bit at a loss here :-(

**Comment by markusdregi** [comment]
> > Observe that it is only the macOS runs that are failing... Maybe that could be a hint?
> 
> Ehhh - no not really; I am a bit at a loss here :-(

That appears to be the situation does it not? At least when I open the travis builds it is only the macOS builds that fails?

**Comment by markusdregi** [comment]
I had an idea about this being related to that `macOS`'s `tmp`-folder really just being a symlink. That has occurred as an issue before. Unfortunately, it did not seem to be the case this time...

**Pull request merged**
